thunder/executors/fa3ex.py

```diff
@@ -0,0 +1,153 @@
+import torch
+
+# flash attn 3
+try:
+    from flash_attn_interface import _flash_attn_forward, _flash_attn_backward, flash_attn_func
+
+    HAS_FA3 = True
+except:
+    HAS_FA3 = False
+
+import thunder
+from thunder.core.transforms import get_grad, put_grads
+from thunder.extend import OperatorExecutor, register_executor
+
+fa3_ex: OperatorExecutor = OperatorExecutor("fa3", version="0.1")
+register_executor(fa3_ex)
+
+
+def fa3_fwd_meta(
+    q: thunder.torch.TensorLike,
+    k: thunder.torch.TensorLike,
+    v: thunder.torch.TensorLike,
+    causal: bool = False,
+    softmax_scale: float | None = None,
+):
+    return thunder.TensorProxy(like=q), thunder.TensorProxy(like=q, shape=(*q.shape[:1], 1))
+
+
+def fa3_fwd_impl(
+    q: torch.Tensor, k: torch.Tensor, v: torch.Tensor, causal: bool = False, softmax_scale: float | None = None
+):
+    if not HAS_FA3:
```

Comment:
Why return `None, None` vs throwing an exception here?

Reply:
Good idea, changed to raise exception